### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by sd-hd**
Hey juju4,

just have time to refactored your branch.

Thank you for the criticism, so i could learn.

### CODE REVIEW

**Comment by juju4** [comment]
Thanks @sd-hd . did some comments. please review.

**Comment by sd-hd** [comment]
@juju4 cannot see your comments ?

**Comment by sd-hd** [comment]
@juju4 cannot find any comments ? did you have some linke for me

File: tasks/lxd-init.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+---
+
+- name: LXD init
+  block:
+  - set_fact:
```
[the remark this answers is not in the document]
Comment (sd-hd) [reply tasks/lxd-init.yml]: true.

but vs-code shows block with an error until i add a name for the block ... ?

File: .kitchen.vagrant.yml [hunk 2]
```diff
@@ -28,9 +28,12 @@ platforms:
   - name: ubuntu-18.04
   - name: ubuntu-16.04
   - name: ubuntu-14.04
-  - name: 'centos/7'
+#  - name: 'centos/7'
```
[the remark this answers is not in the document]
Comment (sd-hd) [reply .kitchen.vagrant.yml]: on my system the kitchen build was broken for centos/7

but i can revert the change

**Comment by sd-hd** [comment]
hey @juju4 
i just fixed some of your comments.

two are still open

i completely new to travis / kitchen, so i need some time to read the doc's 

maybe you can help me out, so that i can learn by example

File: .travis.yml.simple [hunk 3]
```diff
@@ -29,3 +29,4 @@ script:
 # Serverspec
 #  - "test/integration/default/serverspec/run-local-tests.sh"
   - "cd test/integration/default/serverspec/ && bundle exec rake spec"
+  - "cd test/integration/cluster/serverspec/ && bundle exec rake spec"
```
[the remark this answers is not in the document]
Comment (juju4) [reply .travis.yml.simple]: just remove line as duplicate of one above. not sure why file was edited

File: tasks/lxd-init.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (juju4) [reply tasks/lxd-init.yml]: ok. lint is lint and sometimes not obvious catch.
keep name for block but I'm pretty sure a name is needed for set_fact.

**Comment by juju4** [comment]
on serverspec testing, what are the config keywords or command output that you expect to check to validate cluster functional?
that's what you want to put in serverspec tests.

**Comment by sd-hd** [comment]
@juju4 
hope i have fixed all of your comments
could you have a look ?

**Comment by sd-hd** [comment]
@juju4 
refactored again

**Comment by sd-hd** [comment]
@juju4
is there anything i could change too?

**Comment by juju4** [comment]
Just the validation part:
as said previously, on serverspec testing, what are the config keywords or command output that you expect to check to validate cluster functional?
that's what you want to put in serverspec tests.
for reference, https://serverspec.org/resource_types.html
just copy test/integration/default/serverspec/lxd_spec.rb and make a specific test/integration/default/serverspec/lxdcluster_spec.rb for corresponding tests.

Thanks

**Comment by juju4** [comment]
Any comment on how to validate/test cluster functionalities?

**Comment by sd-hd** [comment]
hi juju4,

sorry i did not have the time to implement the ktichen.



did you know someone, who can help me with my ubuntu maas home-lab? i have trouble with deploy, because curtin could not detect my storage...

thank you for merging.

**Pull request merged**